packages/manager/src/features/CloudPulse/shared/CloudPulseDashboardFilterBuilder.tsx

```diff
@@ -0,0 +1,263 @@
+      },
+      [emitFilterChangeByFilterKey]
+    );
+
+    const getProps = React.useCallback(
+      (config: CloudPulseServiceTypeFilters) => {
+        if (config.configuration.filterKey == REGION) {
+          return getRegionProperties(
+            config,
+            handleRegionChange,
+            dashboard,
+            isServiceAnalyticsIntegration
+          );
+        } else if (config.configuration.filterKey == RESOURCE_ID) {
+          return getResourcesProperties(
+            config,
+            handleResourceChange,
+            dashboard,
+            isServiceAnalyticsIntegration,
+            dependentFilterReference.current
+          );
+        } else if (config.configuration.filterKey == RELATIVE_TIME_DURATION) {
+          return getTimeDurationProperties(
+            config,
+            handleTimeRangeChange,
+            isServiceAnalyticsIntegration
+          );
+        } else {
+          return {}; // if the above doesn't match use out custom select for rendering filters, the equivalent component for this will be implemented in upcoming PR's
+        }
+      },
+      [
+        dashboard,
+        handleRegionChange,
+        handleResourceChange,
+        handleTimeRangeChange,
+        isServiceAnalyticsIntegration,
+      ]
+    );
+
+    const toggleShowFilter = () => {
+      setShowFilter((showFilterPrev) => !showFilterPrev);
+    };
+
+    const RenderFilters = React.useCallback(() => {
+      const filters = FILTER_CONFIG.get(dashboard.service_type)?.filters || [];
+
+      if (!filters || filters.length == 0) {
+        // if the filters are not defined , print an error state
+        return (
+          <Grid item key={'filtererror'} xs={12}>
+            <ErrorState
+              CustomIcon={InfoIcon}
+              CustomIconStyles={{ height: '10%', width: '10%' }}
+              errorText={'Please configure filters to continue'}
+            ></ErrorState>
+          </Grid>
+        );
+      }
+
+      return filters
+        .filter((config) =>
+          isServiceAnalyticsIntegration
+            ? config.configuration.neededInServicePage
+            : !config.configuration.neededInServicePage
+        )
+        .map((filter) => (
+          <Grid
+            item
+            key={filter.configuration.filterKey}
+            sx={{ marginLeft: 2 }}
+            xs
+          >
+            {RenderComponent({
+              componentKey: filter.configuration.filterKey,
+              componentProps: { ...getProps(filter) },
+            })}
+          </Grid>
+        ));
+    }, [dashboard, getProps, isServiceAnalyticsIntegration]);
+
+    if (
+      !dashboard ||
+      !dashboard.service_type ||
+      !FILTER_CONFIG.has(dashboard.service_type)
+    ) {
+      return (
+        <ErrorState
+          CustomIcon={InfoIcon}
+          CustomIconStyles={{ height: '10%', width: '10%' }}
+          errorText={'Please pass valid dashboard to render the filters'}
+        ></ErrorState>
```

Comment:
If there are no children, can we use a self-closing tag?

```suggestion
        <ErrorState
          CustomIcon={InfoIcon}
          CustomIconStyles={{ height: '10%', width: '10%' }}
          errorText={'Please pass valid dashboard to render the filters'}
        />
```

Reply:
done

Reply:
done